app/models/spree/gateway/amazon.rb

```diff
@@ -183,6 +183,16 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
+
+      response = @mws.refund(response_code, order.number, payment.credit_allowed, payment.currency)
```

Comment:
I see where you are coming from. Here is the commit: https://github.com/boomerdigital/solidus_amazon_payments/pull/30/commits/af7b0688b4bc86147dcf862538cad32f7c60f65b